### PR TITLE
[3.3.3] lwm2m: add verify trust certificate to bootstrap

### PR DESCRIPTION
**Pull request by nickAS21**
-  add verify trust certificate to bootstrap mode

### CODE REVIEW

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/secure/LwM2mDefaultBootstrapSessionManager.java [hunk 1]
```diff
@@ -83,15 +84,16 @@ public LwM2mDefaultBootstrapSessionManager(BootstrapSecurityStore bsSecurityStor
     @Override
     public BootstrapSession begin(BootstrapRequest request, Identity clientIdentity) {
         boolean authorized = true;
-        Iterator<SecurityInfo> securityInfos;
+        Iterator<SecurityInfo> securityInfos = null;
```
Comment (YevhenBondarenko) [inline common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/secure/LwM2mDefaultBootstrapSessionManager.java:87]: securityInfos initialized below.

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/LwM2MTransportBootstrapService.java [hunk 2]
```diff
@@ -50,6 +52,7 @@
     private final LwM2MBootstrapSecurityStore lwM2MBootstrapSecurityStore;
     private final LwM2MInMemoryBootstrapConfigStore lwM2MInMemoryBootstrapConfigStore;
     private final TransportService transportService;
+    private final TbLwM2MDtlsBootstrapCertificateVerifier certificateVerifier;
```
Comment (YevhenBondarenko) [inline common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/LwM2MTransportBootstrapService.java:55]: Use interface

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/secure/LwM2mDefaultBootstrapSessionManager.java (hunk 1 above)
Comment (nickAS21) [reply common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/secure/LwM2mDefaultBootstrapSessionManager.java:87]: This is used in this^

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/secure/TbLwM2MDtlsBootstrapCertificateVerifier.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+                            }
+                        }
+                        // if not trust or cert trust securityInfo == null
+                        if (securityInfo == null || securityInfo.getMsg() == null) {
+                            String strCert = SslUtil.getCertificateString(cert);
+                            String sha3Hash = EncryptionUtil.getSha3Hash(strCert);
+                            try {
+                                securityInfo = bsSecurityStore.getX509ByEndpoint(sha3Hash);
+                            } catch (LwM2MAuthException e) {
+                                log.trace("Failed find security info: {}", sha3Hash, e);
```
Comment (YevhenBondarenko) [inline common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/secure/TbLwM2MDtlsBootstrapCertificateVerifier.java:131]: Failed to find security info

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/secure/TbLwM2MDtlsBootstrapCertificateVerifier.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+                        }
+                        ValidateDeviceCredentialsResponse msg = securityInfo != null ? securityInfo.getMsg() : null;
+                        if (msg != null && StringUtils.isNotEmpty(msg.getCredentials())) {
+                            x509CredentialsFound = true;
+                            break;
+                        }
+                    } catch (CertificateEncodingException |
+                            CertificateExpiredException |
+                            CertificateNotYetValidException e) {
+                        log.error(e.getMessage(), e);
```
Comment (YevhenBondarenko) [inline common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/secure/TbLwM2MDtlsBootstrapCertificateVerifier.java:142]: Plz prepare the log message

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/store/LwM2MBootstrapSecurityStore.java [hunk 5]
```diff
@@ -68,65 +69,47 @@ public LwM2MBootstrapSecurityStore(EditableBootstrapConfigStore bootstrapConfigS
 
     @Override
     public Iterator<SecurityInfo> getAllByEndpoint(String endPoint) {
```
Comment (YevhenBondarenko) [inline common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/store/LwM2MBootstrapSecurityStore.java:71]: rename to String endpoint

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/store/LwM2MBootstrapSecurityStore.java [hunk 6]
```diff
@@ -68,65 +69,47 @@ public LwM2MBootstrapSecurityStore(EditableBootstrapConfigStore bootstrapConfigS
...
+            TbLwM2MSecurityInfo store = lwM2MCredentialsSecurityInfoValidator.getEndpointSecurityInfoByCredentialsId(endPoint, BOOTSTRAP);
+            this.addValueToStore(store, store.getEndpoint());
+            return store;
+        } catch (LwM2MAuthException e) {
+            log.trace("Failed find security info: {}", endPoint, e);
         }
         return null;
     }
 
+
```
Comment (YevhenBondarenko) [inline common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/store/LwM2MBootstrapSecurityStore.java:112]: remove extra line

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/store/LwM2MBootstrapSecurityStore.java [hunk 7]
```diff
@@ -163,25 +146,23 @@ private LwM2MBootstrapConfig getParametersBootstrap(TbLwM2MSecurityInfo store) {
...
-            }
-            else {
+            } else {
                 if (!lwm2mServerSecurityMode.equals(serverCredential.getSecurityMode())) {
                     validLw.set(false);
                 }
             }
         });
-        return validBs.get()&validLw.get();
+        return validBs.get() & validLw.get();
```
Comment (YevhenBondarenko) [inline common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/store/LwM2MBootstrapSecurityStore.java:165]: & -> &&

File: common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/store/LwM2MBootstrapSecurityStore.java [hunk 8]
```diff
@@ -191,4 +172,47 @@ private boolean getValidatedSecurityMode(LwM2MBootstrapConfig lwM2MBootstrapConf
     public TransportProtos.SessionInfoProto removeSessionByEndpoint(String endpoint) {
         return bsSessions.remove(endpoint);
     }
+
+    public BootstrapConfig getBootstrapConfigByEndpoint(String endpoint) {
+        return bootstrapConfigStore.getAll().get(endpoint);
+    }
+
+    public SecurityInfo addValueToStore(TbLwM2MSecurityInfo store, String endPoint) {
```
Comment (YevhenBondarenko) [inline common/transport/lwm2m/src/main/java/org/thingsboard/server/transport/lwm2m/bootstrap/store/LwM2MBootstrapSecurityStore.java:180]: String endPoint -> String endpoint

**Pull request merged**